common/src/db/mod.rs

```diff
@@ -263,3 +265,37 @@ impl DatabaseErrors for DbErr {
         }
     }
 }
+
+pub trait ColumnsPrefixed {
+    fn columns_prefixed<C, I>(self, prefix: &str, cols: I) -> Self
+    where
+        C: ColumnTrait,
+        I: IntoIterator<Item = C>;
+}
+
+impl<T: QuerySelect> ColumnsPrefixed for T {
+    fn columns_prefixed<C, I>(mut self, prefix: &str, cols: I) -> Self
+    where
+        C: ColumnTrait,
+        I: IntoIterator<Item = C>,
+    {
+        for col in cols.into_iter() {
+            if let SimpleExpr::Column(col_ref) = col.into_simple_expr() {
```

Comment:
To me it looks like that, if the input doesn't match the functions expectation, it will silently be dropped. That might lead to those runtime issues we'd like to avoid.

I think there should at least some some source level doc explaining this. Even better (from my perspective) would be to go for a `try_` style function, returning an error in a case like that.

Reply:
+1